### PR TITLE
V6 upgrade guide

### PR DESCRIPTION
**Pull request by kawikabader**
## Summary

Adds guide for upgrading from V5 to V6.
## Changelog

Added documentation for V6 upgrade.

### CODE REVIEW

File: docs/upgrade-to-6.md [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+make them available on `.config`.
+
+_Impacted Plugins/Packages: `@gasket/plugin-config`_
+
+### Config Moved to Gasket Data
+
+At this time, the [Intl] and [App config] objects are the only config objects
+that have been moved to the `gasketData` object.
+
+_Impacted Plugins/Packages: `@gasket/plugin-config`, `@gasket/plugin-intl`_
```
Comment (kinetifex) [inline docs/upgrade-to-6.md:64]: Let's drop this section. It doesn't actual provide much information and the upgrade details are spelled out elsewhere in the docs

File: docs/upgrade-to-6.md [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+-   "webpack": "^4.44.1",
++   "webpack": "^5.9.0"
+}
+```
+
+### Removed Config Defaults
+
+We have removed the generated Webpack config defaults. The `node` config options
+that were previously prescribed, have changed. You can find more info about
+configuring Node.js options with Webpack 5 on the [webpack website].
```
Comment (kinetifex) [inline docs/upgrade-to-6.md:564]: I have a suggestion branch for reworking this section. See https://github.com/godaddy/gasket/compare/v6-upgrade-guide...v6-upgrade-guide-webpack

You can `cherry-pick` the commit over if you like it.

**Review by kinetifex — COMMENTED** [review]
Looks good. I committed my sections, ran it through Grammarly, and tweaked some wording here and there. See my suggested rework of the Webpack 5 section.

**Review by kinetifex — APPROVED** [review]

**Review by burgessa23 — APPROVED** [review]

File: docs/upgrade-to-6.md [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+-   "next-redux-wrapper": "^4.0.1",
++   "next-redux-wrapper": "^6.0.2",
++   "lodash.merge": "^4.6.2",
+    "@gasket/plugin-redux": "^6.0.0"
+    ...
+  }
+}
+```
+
+You will also want a deep merge utility to help set up a basic HYDRATE actual
```
Comment (rxmarbles) [inline docs/upgrade-to-6.md:132]:
```suggestion
You will also want a deep merge utility to help set up a basic HYDRATE
```

**Review by rxmarbles — APPROVED** [review]
Just a nit

**Review by mswaagman-godaddy — APPROVED** [review]

**Pull request merged**